pages/deployment/[uuid].tsx

```diff
@@ -364,35 +297,55 @@ export default function Deployment() {
       </Wrapper>
       <Dialog maxWidth='lg' onClose={handleClose} open={open}>
         <DialogTitle sx={{ backgroundColor: theme.palette.container.main }}>Pull from Docker</DialogTitle>
-        <DialogContent>
-          <DialogContentText sx={{ p: 2 }}>
-            <Box>
-              <p style={{ margin: 0 }}>
-                # Login to Docker (your token can be found on the
-                <Link href='/settings' passHref>
-                  <MuiLink sx={{ ml: 0.5, mr: 0.5, color: theme.palette.secondary.main }}>settings</MuiLink>
-                </Link>
-                page)
-              </p>
-              <CodeLine line={`docker login ${uiConfig.registry.host} -u ${currentUser.id}`} />
-              <br />
-
-              <p style={{ margin: 0 }}># Pull model</p>
-              <CodeLine line={`docker pull ${deploymentTag}`} />
-              <br />
-
-              <p style={{ margin: 0 }}># Run Docker image</p>
-              <CodeLine line={`docker run -p 9000 ${deploymentTag}`} />
-              <br />
-
-              <p style={{ margin: 0 }}># Check that the Docker container is running</p>
-              <CodeLine line='docker ps' />
-              <br />
-
-              <p style={{ margin: 0 }}># The model is accessible at localhost:9000</p>
-            </Box>
-          </DialogContentText>
-        </DialogContent>
+        {versions && (
+          <DialogContent>
+            <Stack justifyContent='flex-start' alignItems='center' direction='row' spacing={2} sx={{ m: 2 }}>
+              <Typography>Select a version</Typography>
+              <Select value={selectedImageTag} label='Selected version' onChange={onSelectedTagChange}>
+                {versions
```

Comment:
Unlikely to be a problem, but if we had a lot of versions then doing multiple array manipulations on ever render is more expensive than needed. Consider pulling this filter and map into a `useMemo`